client/app/cloud/cloud.constants.js

```diff
@@ -48,6 +48,235 @@ angular.module('managerApp')
     'g2',
     'g3',
   ])
+  .constant('CLOUD_FLAVOR_TYPES', {
```

Comment:
Shouldn't it be easiest to have a method that handle that?
A method with following rules :
* if flavor is suffixed by `-flex` remove the suffix and returns the flavor to uppercase and add ' - Flex' ;
* if flavor is prefixed by `win-` remove the prefix and return the flavor to uppercase ;
* otherwise return the flavor to uppercase.

If tomorrow, a new flavor is added, we may be lucky to have nothing to do :-)

What do you think of this solution?

Reply:
You could do this:
```js
const rules = [
        {
          condition: ({ currentIndex, currentValue }) => currentIndex === 0 && `${currentValue}`.toUpperCase() === 'win',
          mapping: () => '',
        },
        {
          condition: ({ currentValue, inputParts, currentIndex }) => `${currentValue}`.toUpperCase() === 'FLEX' && inputParts.length - 1 === currentIndex,
          mapping: value => ` - ${value[0].toUpperCase()}${value.substr(1)}`,
        },
        {
          condition: () => true,
          mapping: value => `${value.toUpperCase()}`,
          intermediateHyphen: true,
        },
      ];

function turnIntoReadable(datum) {
const inputParts = datum.raw.split('-');

        return [].concat(...inputParts
          .map((currentValue, currentIndex) => {
            const matchingRule = rules.find(rule => rule.condition({
              currentValue,
              currentIndex,
              inputParts,
            }));

            const rawReturnValue = matchingRule.mapping(currentValue);
            return matchingRule.intermediateHyphen
              ? [{ value: rawReturnValue, needsHyphen: true }, { isHyphen: true }]
              : { value: rawReturnValue };
          }))
          .reduce((accumulator, currentValue, currentIndex, array) => {
            let valueToAppend = '';
            if (currentValue.isHyphen) {
              if (currentIndex + 1 !== array.length) {
                valueToAppend = '-';
              }
            }

            if (!currentValue.isHyphen && currentValue.needsHyphen) {
              valueToAppend = currentValue.value;
            }

            if (!currentValue.isHyphen && !currentValue.needsHyphen) {
              accumulator.pop();
              valueToAppend = currentValue.value;
            }

            return [...accumulator, valueToAppend];
          }, [])
          .join('');
}
```